File: smartstore_backend/src/main/java/com/ssafy/smartstore/model/service/CommentService.java
package com.ssafy.smartstore.model.service;

import com.ssafy.smartstore.model.dto.Comment;
import java.util.List;

public interface CommentService {
    /**
     * Comment를 등록한다.
     * @param comment
     */
    void addComment(Comment comment);

    /**
     * id에 해당하는 comment를 반환한다.
     * @param id
     * @return
     */
    Comment selectComment(Integer id);

    /**
     * id에 해당하는 Comment를 삭제한다.
     * @param id
     */
    void removeComment(Integer id);

    /**
     * Comment를 수정한다. 수정 내용은 rating과 comment이다.
     * @param comment
     */
    void updateComment(Comment comment);

    /**
     * productId에 해당하는 Comment의 목록을 Comment id의 내림차순으로 반환한다.
     * @param productId
     * @return
     */
    List<Comment> selectByProduct(Integer productId);
}

File: Smartstore_Network-starter/app/src/main/java/com/ssafy/smartstore/service/CommentService.kt
package com.ssafy.smartstore.service


import com.ssafy.smartstore.dto.Comment
import com.ssafy.smartstore.util.RetrofitCallback
import com.ssafy.smartstore.util.RetrofitUtil
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

private const val TAG = "CommentService_싸피"
class CommentService {

    fun addComment(comment: Comment, callback: RetrofitCallback<Boolean>) {
        RetrofitUtil.commentService.insert(comment).enqueue(object : Callback<Boolean> {
            override fun onResponse(call: Call<Boolean>, response: Response<Boolean>) {
                val res = response.body()
                if (response.isSuccessful) {
                    if (res != null) {
                        callback.onSuccess(response.code(), res)
                    }
                } else {
                    callback.onFailure(response.code())
                }
            }

            override fun onFailure(call: Call<Boolean>, t: Throwable) {
                callback.onError(t)
            }

        })
    }

    fun updateComment(comment: Comment, callback: RetrofitCallback<Boolean>) {
        RetrofitUtil.commentService.update(comment).enqueue(object : Callback<Boolean> {
            override fun onResponse(call: Call<Boolean>, response: Response<Boolean>) {
                val res = response.body()
                if (response.isSuccessful) {
                    if (res != null) {
                        callback.onSuccess(response.code(), res)
                    }
                } else {
                    callback.onFailure(response.code())
                }
            }

            override fun onFailure(call: Call<Boolean>, t: Throwable) {
                callback.onError(t)
            }

        })
    }

    fun removeComment(id: Int, callback: RetrofitCallback<Boolean>) {
        RetrofitUtil.commentService.delete(id).enqueue(object : Callback<Boolean> {
            override fun onResponse(call: Call<Boolean>, response: Response<Boolean>) {
                val res = response.body()
                if (response.isSuccessful) {
                    if (res != null) {
                        callback.onSuccess(response.code(), res)
                    }
                } else {
                    callback.onFailure(response.code())
                }
            }

            override fun onFailure(call: Call<Boolean>, t: Throwable) {
                callback.onError(t)
            }

        })
    }
}